nixos/modules/programs/x2goserver.nix

```diff
@@ -0,0 +1,121 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.x2goserver;
+
+  defaults = {
+    log = { "loglevel" = cfg.logLevel; };
+    x2goagent = { "port_randomization" = "pure-random"; };
+    superenicer = { "enable" = "no"; };
+    telekinesis = { "enable" = "no"; };
+  };
+  confText = generators.toINI {} (recursiveUpdate defaults cfg.config);
+  x2goServerConf = pkgs.writeText "x2goserver.conf" confText;
+
+  x2goAgentOptions = pkgs.writeText "x2goagent.options" ''
+    X2GO_NXOPTIONS=""
+    X2GO_NXAGENT_DEFAULT_OPTIONS="${concatStringsSep " " cfg.nxagentDefaultOptions}"
+  '';
+
+in {
+  options.programs.x2goserver = {
+    enable = mkEnableOption "x2goserver";
+
+    logLevel = mkOption {
+      type = types.str;
+      default = "notice";
+      description = ''
+        The log level x2goserver will log at.
+        Possible levels are: emerg, alert, crit, err, warning, notice, info, debug
+      '';
+    };
+
+    nxagentDefaultOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ "-extension GLX" "-nolisten tcp" ];
```

Comment:
Where does this default come from?

Reply:
It's default in the config, the GLX option was necessary for me to get some apps to run.